core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -142,16 +142,28 @@ ArtifactResultBuildItem nativeSourcesResult(NativeConfig nativeConfig,
                 .setNativeImageFeatures(nativeImageFeatures)
                 .build();
         List<String> command = nativeImageArgs.getArgs();
-        try (FileOutputStream commandFOS = new FileOutputStream(outputDir.resolve("native-image.args").toFile())) {
+        try (FileOutputStream commandFOS = new FileOutputStream(outputDir.resolve("native-image.args").toFile());
+                FileOutputStream dockerImageNameFOS = new FileOutputStream(outputDir.resolve("graalvm.version").toFile())) {
             String commandStr = String.join(" ", command);
             commandFOS.write(commandStr.getBytes(StandardCharsets.UTF_8));
 
-            log.info("The sources for a subsequent native-image run along with the necessary arguments can be found in "
-                    + outputDir);
+            dockerImageNameFOS.write(GraalVM.Version.CURRENT.version.toString().getBytes(StandardCharsets.UTF_8));
         } catch (Exception e) {
             throw new RuntimeException("Failed to build native image sources", e);
         }
 
+        if (nativeConfig.isContainerBuild()) {
+            try (FileOutputStream dockerImageNameFOS = new FileOutputStream(
+                    outputDir.resolve("native-builder.image").toFile())) {
+                dockerImageNameFOS.write(nativeConfig.getEffectiveBuilderImage().getBytes(StandardCharsets.UTF_8));
```

Comment:
Maybe use `Files.write()` instead? And we can do the same above?

A simple try/catch with all 3 Files.write() should be the best outcome?

You can do it in a separate commit given it's a refactoring of an existing code.